build_scripts/pack/generate-builder-for-project.sh

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env bash
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+if [[ "$#" -ne 2 ]]; then
+  echo 2> "Usage: $(basename "$0") <project-id> <destination-directory>"
+  exit 1
+fi
+
+readonly PROJECT_ID="${1}"
+readonly DEST="${2}"
+
+mkdir -p "${DEST}/buildpack/bin"
+
+cat >"${DEST}/builder.toml" <<_EOF_
+[[buildpacks]]
+uri = "buildpack"
+
+[[order]]
+    [[order.group]]
+    id = "com.google.buildpack.cpp.project"
+    version = "0.4.0"
+
+# Stack that will be used by the builder
+[stack]
+id = "google"
+run-image   = "gcr.io/${PROJECT_ID}/functions-framework-cpp/run-image:latest"
+build-image = "gcr.io/${PROJECT_ID}/functions-framework-cpp/build-image:latest"
+_EOF_
+
+cat >"${DEST}/buildpack/buildpack.toml" <<_EOF_
+api = "0.2"
+
+[buildpack]
+id = "com.google.buildpack.cpp.project"
+version = "0.4.0"
+name = "Functions Framework C++ Buildpack for GCP Project ${PROJECT_ID}"
+
+[[stacks]]
+id = "google"
+_EOF_
+
+cat >"${DEST}/buildpack/bin/detect" <<'_EOF_'
```

Comment:
Do we need to ensure that the "#!/bin/bash" files are executable?

Reply:
Evidently they work either way, but seems nicer to make them `755`.